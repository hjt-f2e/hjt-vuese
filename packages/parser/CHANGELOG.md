# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [2.4.3](https://github.com/vuese/vuese/compare/@hjtvuese/parser@2.4.2...@hjtvuese/parser@2.4.3) (2019-10-22)


### Bug Fixes

* switched to isNumericLiteral(â€¦) ([#117](https://github.com/vuese/vuese/issues/117)) ([96cfa67](https://github.com/vuese/vuese/commit/96cfa67))
