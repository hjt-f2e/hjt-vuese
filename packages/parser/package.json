{
  "name": "@hjtvuese/parser",
  "version": "2.10.9",
  "description": "Vue file parser for automatic document generation",
  "keywords": [
    "parser",
    "vue",
    "document"
  ],
  "author": "young.zhou <small_yangy@foxmail.com>",
  "homepage": "https://github.com/hjt-f2e",
  "license": "MIT",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "dependencies": {
    "@babel/generator": "^7.12.10",
    "@babel/parser": "^7.12.0",
    "@babel/traverse": "^7.12.0",
    "@babel/types": "^7.3.4",
    "vue-template-compiler": "^2.6.11"
  },
  "scripts": {
    "tsc": "rimraf dist && tsc"
  },
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/hjt-f2e/hjt-vuese.git"
  },
  "bugs": {
    "url": "https://github.com/hjt-f2e/hjt-vuese/issues"
  },
  "buildOptions": {
    "extractDts": true
  },
  "gitHead": "2bc1b0691aaf9480fa6fc0ad59ff1bc991b81efe"
}
