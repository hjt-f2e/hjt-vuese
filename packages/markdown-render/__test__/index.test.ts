import Render, { RenderResult, MarkdownResult } from '@hjtvuese/markdown-render'
import { ParserResult } from '@hjtvuese/parser'

test('Proper rendering of the table header', () => {
  const res: ParserResult = {
    name: 'MyComponent',
    componentDesc: {
      default: ['This is a description of the component'],
      group: ['My Group']
    },
    props: [
      {
        name: 'someProp',
        type: ['String'],
        typeDesc: ['`TOP` / `BOTTOM`'],
        required: true,
        defaultDesc: ['`TOP`'],
        describe: ['Represents the direction of the arrow']
      }
    ],
    events: [
      {
        name: 'click',
        isSync: false,
        syncProp: '',
        describe: ['Triggered when clicked'],
        argumentsDesc: ['a boolean value']
      }
    ],
    slots: [
      {
        name: 'header',
        describe: 'Table header',
        backerDesc: '`<th>{{title}}</th>`',
        bindings: {},
        scoped: false,
        target: 'template'
      },
      {
        name: 'header',
        describe: 'Table header',
        backerDesc: '`<th>{{title}}</th>`',
        bindings: {},
        scoped: false,
        target: 'script'
      }
    ],
    methods: [
      {
        name: 'clear',
        describe: ['Clear form'],
        argumentsDesc: ['a boolean value']
      }
    ],
    mixIns: [
      {
        mixIn: 'MixIn'
      }
    ],
    data: [
      {
        name: 'someVariable',
        type: 'String',
        describe: ['Does something'],
        default: ''
      }
    ],
    watch: [
      {
        name: 'question',
        describe: ['Something about a question'],
        argumentsDesc: ['a string value']
      }
    ]
  }
  const render = new Render(res)
  const renderRes: RenderResult = render.render()
  const markdownRes = render.renderMarkdown() as MarkdownResult
  expect(renderRes).toMatchSnapshot()
  expect(markdownRes).toMatchSnapshot()
})

describe('Empty Components (without slots,methods,props,events) can be forced to be rendered', () => {
  test('Does not render an empty component by defaut', () => {
    const res: ParserResult = {
      name: 'MyComponent',
      componentDesc: {
        default: ['This is a description of the component']
      }
    }
    const render = new Render(res)
    const renderRes: RenderResult = render.render()
    const markdownRes = render.renderMarkdown() as MarkdownResult
    expect(markdownRes).toBeNull()
    expect(renderRes).toEqual({})
  })
  test('Does render an empty component by with @hjtvuese in the description', () => {
    const res: ParserResult = {
      name: 'MyComponent',
      componentDesc: {
        default: [
          'This is a description of the component.',
          'Supplementary description'
        ],
        vuese: ['']
      }
    }
    const render = new Render(res)
    const renderRes: RenderResult = render.render()
    const markdownRes = render.renderMarkdown() as MarkdownResult
    expect(markdownRes).not.toBeNull()
    expect(markdownRes).toMatchSnapshot()
    expect(renderRes).toMatchSnapshot()
  })
})
