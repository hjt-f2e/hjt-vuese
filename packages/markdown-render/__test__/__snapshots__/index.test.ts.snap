// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty Components (without slots,methods,props,events) can be forced to be rendered Does render an empty component by with @hjtvuese in the description 1`] = `
Object {
  "componentName": "MyComponent",
  "content": "# MyComponent

This is a description of the component. Supplementary description

",
  "groupName": "BASIC",
}
`;

exports[`Empty Components (without slots,methods,props,events) can be forced to be rendered Does render an empty component by with @hjtvuese in the description 2`] = `Object {}`;

exports[`Proper rendering of the table header 1`] = `
Object {
  "data": "|Name|Type|Description|Default|
|---|---|---|---|
|someVariable|\`String\`|Does something|-|
",
  "events": "|Event Name|Description|Parameters|
|---|---|---|
|click|Triggered when clicked|a boolean value|
",
  "methods": "|Method|Description|Parameters|
|---|---|---|
|clear|Clear form|a boolean value|
",
  "mixIns": "|MixIn|
|---|
|MixIn|
",
  "props": "|Name|Description|Type|Required|Default|
|---|---|---|---|---|
|someProp|Represents the direction of the arrow|\`TOP\` / \`BOTTOM\`|\`true\`|\`TOP\`|
",
  "slots": "|Name|Description|Default Slot Content|
|---|---|---|
|header|Table header|\`<th>{{title}}</th>\`|
",
  "watch": "|Name|Description|Parameters|
|---|---|---|
|question|Something about a question|a string value|
",
}
`;

exports[`Proper rendering of the table header 2`] = `
Object {
  "componentName": "MyComponent",
  "content": "# MyComponent

This is a description of the component

## Props

<!-- @hjtvuese:MyComponent:props:start -->
|Name|Description|Type|Required|Default|
|---|---|---|---|---|
|someProp|Represents the direction of the arrow|\`TOP\` / \`BOTTOM\`|\`true\`|\`TOP\`|

<!-- @hjtvuese:MyComponent:props:end -->


## Events

<!-- @hjtvuese:MyComponent:events:start -->
|Event Name|Description|Parameters|
|---|---|---|
|click|Triggered when clicked|a boolean value|

<!-- @hjtvuese:MyComponent:events:end -->


## Slots

<!-- @hjtvuese:MyComponent:slots:start -->
|Name|Description|Default Slot Content|
|---|---|---|
|header|Table header|\`<th>{{title}}</th>\`|

<!-- @hjtvuese:MyComponent:slots:end -->


## Methods

<!-- @hjtvuese:MyComponent:methods:start -->
|Method|Description|Parameters|
|---|---|---|
|clear|Clear form|a boolean value|

<!-- @hjtvuese:MyComponent:methods:end -->


## MixIns

<!-- @hjtvuese:MyComponent:mixIns:start -->
|MixIn|
|---|
|MixIn|

<!-- @hjtvuese:MyComponent:mixIns:end -->


## Data

<!-- @hjtvuese:MyComponent:data:start -->
|Name|Type|Description|Default|
|---|---|---|---|
|someVariable|\`String\`|Does something|-|

<!-- @hjtvuese:MyComponent:data:end -->


## Watch

<!-- @hjtvuese:MyComponent:watch:start -->
|Name|Description|Parameters|
|---|---|---|
|question|Something about a question|a string value|

<!-- @hjtvuese:MyComponent:watch:end -->


",
  "groupName": "My Group",
}
`;
